docs/source/contributor-guide.rst

```diff
@@ -39,39 +47,34 @@ We do not specify any particular schema for the output but we do recommend best
     Deviating from already existing formats complicates modifications to a parser.
 
 
-Beyond recommendations about the data type, we have a recommendations for the parser behavior:
-
-#. *Fail loudly and as specifically as possible*
-    Parsers should return exceptions when provided incompatible data and not fail by returning data.
-    Provide an error message that indicates whether the file is incompatible, or the parser is misconfigured (e.g., missing a required library).
+We also have a recommendations for the parser behavior:
 
 #. *Avoid configuration options that change only output format*
     Parsers can take configuration options that alter the output format, but configurations should be used sparingly.
     A good use of configuration would be to disable complex parsing operations if unneeded.
     A bad use of configuration would be to change the output to match a different schema.
     Operations that significantly alter the form but not the content of a summary should be implemented as adaptors.
 
-#. *Consider whether context should be configuration.
+#. *Consider whether context should be configuration*
     Settings that are identical for each file could be better suited as configuration settings than as context.
 
-Implementing ``is_valid``
--------------------------
-
-The ``is_valid`` operation is used to determine whether a set of files are compatible with the parser.
-Implementing it is optional and recommended only when there are fast ways of testing file types.
-The default implementation is to call ``parse`` and see if the parser fails.
-``is_valid`` should return ``False`` for incompatible files and only throw Exceptions if the parser is misconfigured.
-
-
 Implementing ``group``
 ----------------------
 
-The ``group`` operation finds all groups of files in a directly that should be treated as related units.
+The ``group`` operation finds all groups of files in a directory that should be parsed together.
 Implementing ``group`` is optional.
 The default implementation is to return each file in the directory as its own group.
-New implementations of ``group`` need not return every file as part of a group (e.g., it can perform compatibility filtering)
-and files are allowed to appear in more than 1 group.
-In general, our recommendation is to return all possible groupings of files when in doubt and let the ``is_valid`` and ``parse`` operations sort out which groupings are valid.
+
+Files are allowed to appear in more than one group,
+and we recommend generating all possible groupings of files when in doubt.
```

Comment:
This line sounds like the grouper is encouraged to maximize the number of groups, which is going to lead to duplicate metadata units, or that the grouper should return all permutations of a file group, which I don't think is the intention. (For example, a VASP calculation can be parsed with only an `OUTCAR`, or with an `OUTCAR` and `INCAR`, or with an `OUTCAR` and `WAVCAR`, etc., but the grouper should make one group: `OUTCAR` + all other `*CAR` files in that directory.)
Perhaps the recommendation should be to generate the largest valid group (regardless of if files have been used in another group) instead of all possible groupings?

Reply:
I agree, we should change to recommending "largest valid group"